packages/plugins/type-designer/src/components/drawer-content.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+// CONSTANTS
+import { SCD_ELEMENTS } from '@oscd-plugins/core'
+// SMUI COMPONENTS
+import Textfield from '@smui/textfield'
+import Button from '@smui/button'
+// STORES
+import { pluginStore } from '@/stores/plugin.store'
+import { drawerStore } from '@oscd-plugins/ui'
+// TYPES
+import type { DataTypeTemplates } from '@oscd-plugins/core'
+
+//====== INITIALIZATION ====//
+
+// props
+export let typeElement: DataTypeTemplates.TypeElement
+export let currentColumn: keyof typeof SCD_ELEMENTS
+
+let localAttributes: Record<(typeof standardAttributes)[number], string>
+
+//====== REACTIVITY ====//
+$: ({ element, ...attributes } = typeElement)
+$: standardAttributes = SCD_ELEMENTS[currentColumn]?.element.standardAttributes
+$: setNewLocalAttributes(attributes)
+
+//====== FUNCTIONS =====//
+
+function updateTypeElement() {
+	pluginStore.createAndDispatchUpdateActionEvent({
+		element,
+		oldAttributes: attributes,
+		newAttributes: localAttributes
+	})
+	drawerStore.handleCloseDrawer()
+}
+
+function setNewLocalAttributes(
```

Comment:
From a backend dev's point of view, i see a risk of [race condition](https://www.techtarget.com/searchstorage/definition/race-condition#:~:text=A%20race%20condition%20is%20an,sequence%20to%20be%20done%20correctly.)

If my concern is correct, there may be a brief moment when local and new attributes collide because both are being written to simultaneously.

1. The spread operation creates a new object each time
1. If multiple updates happen in a brief time period, we might lose our updates because of the sveltes reactivity

Just to be sure - are we 100% sure that between the time we read localAtts. and write back the spread result, no reactive updates from the store will modify them?

Reply:
This has little chance to happen in my opinion, but is still a good point and we might keep it in mind for future improvements.